firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java

```diff
@@ -107,34 +105,12 @@ private boolean isDeviceAllowedToSendNetworkEvents() {
   }
 
   /**
-   * Check if we should log the {@link PerfMetric} to transport.
-   *
-   * <p>Cases in which we don't log a {@link PerfMetric} to transport:
-   *
-   * <ul>
-   *   <li>It is a {@link TraceMetric}, the {@link PerfSession} is not verbose and trace metrics are
-   *       sampled.
-   *   <li>It is a {@link NetworkRequestMetric}, the {@link PerfSession} is not verbose and network
-   *       requests are sampled.
-   *   <li>The number of metrics being sent exceeds what the rate limiter allows.
-   * </ul>
+   * Check if the {@link PerfMetric} should be rate limited.
    *
    * @param metric {@link PerfMetric} object.
    * @return true if allowed, false if not allowed.
    */
-  boolean check(PerfMetric metric) {
-    if (metric.hasTraceMetric()
-        && !isDeviceAllowedToSendTraces()
-        && !hasVerboseSessions(metric.getTraceMetric().getPerfSessionsList())) {
-      return false;
-    }
-
-    if (metric.hasNetworkRequestMetric()
-        && !isDeviceAllowedToSendNetworkEvents()
-        && !hasVerboseSessions(metric.getNetworkRequestMetric().getPerfSessionsList())) {
-      return false;
-    }
-
+  boolean checkDeviceRateLimiting(PerfMetric metric) {
```

Comment:
Nit: Can you rename it as `isDeviceRateLimited`? Methods return boolean are suggested to have names that begin with `is` or `has`.

Reply:
Done